site/content/entry/citra-progress-report-2017-september.md

```diff
@@ -0,0 +1,360 @@
+slow, a handful do use it for a handful of visual effects, like New Super Mario
+Bros. 2.
+
+About a year ago when the GPU code on Citra was rewritten, a handful of calls to
+the drawing routine were removed, as it was believed they were unnecessary. Turns
+out, one of the calls was actually needed for some effects in games, as it handled
+immediate mode drawing. This wasn't noticed for a very long time, as most games
+appeared to carry on with no side-effects at all from the rewrite, but was eventually
+found after some research courtesy of [ds84182](https://github.com/ds84182).
+
+## [Interpolate audio samples on a frame-by-frame basis](https://github.com/citra-emu/citra/pull/2858) by [MerryMage](https://github.com/MerryMage)
+
+When a 3DS game needs some sort of audio processing, they can access the 3DS' DSP,
+or __D__igital __S__ound __P__rocessor. It's another processor, alongside the ARM9
+and ARM11, that is given a firmware to run, which in turn is given a bunch of audio
+samples and parameters by the game. The DSP then plays back the buffer in chunks
+of about 5 milliseconds. Each one of these chunks is called an audio frame.
+
+As of today, we don't know how the DSP exactly works, and we don't know how any
+of the firmwares exactly work. (Did I forget to mention earlier there's multiple
+versions of the firmware?) But we do know how to use it, and from there we can
+reimplement its behaviour directly in Citra. Which is exactly what [MerryMage](https://github.com/MerryMage)
+did, which in turn brought audio support for the first time in Citra.
+
+This approach, although having the advantages of being easier to implement, easier
+to understand in code, and has a higher potential of being faster, it has the
+disadvantage that accuracy suffers significantly, especially when shortcuts are
+taken for the sake of speed. One of these shortcuts was in the audio interpolation,
+which is a way of inferring more audio samples from relatively very few existing
+samples.
+
+On a real 3DS, games are allowed to interpolate different audio frames with
+different functions, even in they're in the same buffer. On the other hand, Citra
+interpolated the entire buffer with one function as soon as it was loaded. This
+led to various effects and music in games to sound strange or inaccurate in some way.
+
+One example of this is Deku Link's footsteps in *The Legend of Zelda: Majora's Mask 3D*.
+
+Here's the output of a real 3DS console, for reference:
+
+{{< audio src="/images/entry/citra-progress-report-2017-september/deku-hardware.ogg" >}}
+
+And here's the output of Citra, before this was fixed:
+
+{{< audio src="/images/entry/citra-progress-report-2017-september/deku-pre2858.ogg" >}}
+
+Now that it's been fixed, his footsteps sound a lot better:
+
+{{< audio src="/images/entry/citra-progress-report-2017-september/deku-post2858.ogg" >}}
+
+Audio emulation in Citra is still woefully inaccurate for now, though
```

Comment:
Might be better to drop "woefully", since it's 100% correct in some of the games our users most care about